### PR TITLE
Issue #179295 task: Add webhook support for Shika/Vowel in TjNotifications

### PR DESCRIPTION
**Pull request by praneettekdi**

### CODE REVIEW

File: src/com_tjnotifications/admin/models/forms/webhookfields.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			type="contentlanguage"
+			label="JFIELD_LANGUAGE_LABEL"
+			description="COM_TJNOTIFICATIONS_NOTIFICATION_LANGUAGE_DESC"
+			required="true"
+			>
+			<option value="*">JALL</option>
+		</field>
+
+		<field
+			name="use_gbwh_url"
```
Comment (manojLondhe) [inline src/com_tjnotifications/admin/models/forms/webhookfields.xml:19]: lets indicate clearly as
use_gbwh_url => use_global_webhook_url

File: src/com_tjnotifications/admin/models/notification.php [hunk 2]
```diff
@@ -391,6 +397,26 @@ public function save($data)
 			// 2.2 Find existing template config entries to be deleted (i.e. language specific templates removed by user)
 			foreach ($data[$backend][$backend . 'fields'] as $backendName => $backendFieldValues)
 			{
+				// webhook stuff starts here
```
Comment (manojLondhe) [inline src/com_tjnotifications/admin/models/notification.php:400]: phpcs

// webhook => // Webhook

File: src/com_tjnotifications/admin/models/notification.php [hunk 3]
```diff
@@ -391,6 +397,26 @@ public function save($data)
...
+					}
+					// If using global webhook URl & the global urls are empty
+					elseif ($backendFieldValues['use_gbwh_url'] && empty($webhookUrls[0]))
+					{
+						$this->setError(Text::_('COM_TJNOTIFICATIONS_TEMPLATE_ERR_MSG_GLOBAL_WEBHOOK_URLS'));
+
+						return false;
+					}
+				}
+				// webhook stuff ends here
```
Comment (manojLondhe) [inline src/com_tjnotifications/admin/models/notification.php:418]: phpcs

// webhook => // Webhook

File: src/com_tjnotifications/admin/sql/install.mysql.utf8.sql [hunk 4]
```diff
@@ -28,6 +28,8 @@ CREATE TABLE IF NOT EXISTS `#__tj_notification_template_configs` (
   `language` char(7) NOT NULL DEFAULT '*',
   `subject` text NOT NULL,
   `body` text NOT NULL,
+  `webhook_url` text DEFAULT NULL,
+  `use_gbwh_url` TINYINT(1) NOT NULL DEFAULT '1' COMMENT 'Use Global Config Webhook URLs',
```
Comment (manojLondhe) [inline src/com_tjnotifications/admin/sql/install.mysql.utf8.sql:32]: rename as said above, everywhere

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/com_tjnotifications/admin/language/en-GB.com_tjnotifications.ini [hunk 5]
```diff
@@ -286,3 +286,27 @@ COM_TJNOTIFICATIONS_NOTIFICATION_SMS_REMAINING_EXCEEDED="characters exceeded"
 ; Notification - SMS provider template ID
 COM_TJNOTIFICATIONS_FIELD_SMS_TEMPLATE_ID_LABEL="SMS template ID"
 COM_TJNOTIFICATIONS_FIELD_SMS_TEMPLATE_ID_DESC="Set template ID which is proided by SMS provider"
+
+; Since __DEPLOY_VERSION__
+; Notification - Webhook support
+COM_TJNOTIFICATIONS_FIELD_WEBHOOK_URL_SUBFORM_LABEL="Add Custom Webhook Urls"
```
[the remark this answers is not in the document]
Comment (praneettekdi) [reply src/com_tjnotifications/admin/language/en-GB.com_tjnotifications.ini:292]: ok

**Pull request merged**